ls1intum/Artemis
# `Development`: Use default value if exercise has no plagiarism checks config

### Checklist
#### General

#### Client

### Motivation and Context
Importing programming exercises from files created before merging of the continuous plagiarism control was not possible, because such files didn't contain plagiarism checks config. This caused the client to load the exercise creation form incorrectly. 

Detected during tests of: #7624

Server part fixed in: #7640

### Description
This PR adds a nullability check for exercise's plagiarism detection config before the exercise form is initialised. If exercise does not contain plagiarism detection config, the component inserts a default value into it.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 zip file with Programming Exercise without plagiarism detection config. You can use this one: [Material-cr1-Sorting_Algorithms_Implementation-52-20231125-105137.zip](https://github.com/ls1intum/Artemis/files/13463640/Material-cr1-Sorting_Algorithms_Implementation-52-20231125-105137.zip)

1. Go to course -> exercises.
2. Start programming exercise from the file provided.
3. Verify that the form displays correctly (including default plagiarism detection config).
4. Import the exercise.
5. Verify that the exercise was imported correctly. 

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| exercise-update-plagiarism-component.ts | 63% (testing just the new method)| ✅              |
